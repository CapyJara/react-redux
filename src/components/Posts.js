import React from 'react';
import PropTypes from 'prop-types';
import Post from './Post';

function Posts({ postArr }) {
  console.log('postArr', postArr);

  const postList = postArr.map(post => (

    <li key={post.id}>
      <Post post={post} />
    </li>
  ));

  console.log('postList', postList);

  return (
    <ul>
      {postList}
    </ul>
  );
}

Posts.propTypes = {
  postArr: PropTypes.array.isRequired
};

export default Posts;
