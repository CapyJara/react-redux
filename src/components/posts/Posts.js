import React from 'react';
import PropTypes from 'prop-types';
import Post from './Post';


function Posts({ posts }) {
  const postList = posts.map((post, i) => (

    <li key={`${post.title}-${i}`}>
      <Post post={post} />
    </li>
  ));

  return (
    <ul>
      {postList}
    </ul>
  );
}

Posts.propTypes = {
  posts: PropTypes.array.isRequired
};

export default Posts;
