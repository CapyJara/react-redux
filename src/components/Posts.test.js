import React from 'react';
import { shallow } from 'enzyme';
import Posts from './Posts';

describe('Posts component', () => {
  it('renders a Posts', () => {
    const postArr = [
      { title: 'title1', postBody: 'things about things' },
      { title: 'title2', postBody: 'things' },
      { title: 'title3', postBody: 'things things' },
      { title: 'title4', postBody: 'about' }
    ];
    const wrapper = shallow(<Posts postArr={postArr}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
