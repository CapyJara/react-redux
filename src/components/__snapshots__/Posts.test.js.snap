// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component renders a Posts 1`] = `
<ul>
  <li>
    <Post
      post={
        Object {
          "postBody": "things about things",
          "title": "title1",
        }
      }
    />
  </li>
  <li>
    <Post
      post={
        Object {
          "postBody": "things",
          "title": "title2",
        }
      }
    />
  </li>
  <li>
    <Post
      post={
        Object {
          "postBody": "things things",
          "title": "title3",
        }
      }
    />
  </li>
  <li>
    <Post
      post={
        Object {
          "postBody": "about",
          "title": "title4",
        }
      }
    />
  </li>
</ul>
`;
